modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1353,15 +1410,159 @@ def fold(self, axis, func):
+
+        Notes
+        -----
+        During the join, this dataframe is considered the left, while the other is
+            treated as the right.
+
+        Only inner joins, left outer, right outer, and full outer joins are currently supported.
+            Support for other join types (e.g. natural join) may be implemented in the future.
+        """
+        pass
+
+    def rename(
+        self,
+        new_row_labels: Optional[Union[Dict[Hashable, Hashable], Callable]] = None,
+        new_col_labels: Optional[Union[Dict[Hashable, Hashable], Callable]] = None,
+        level: Optional[Union[int, List[int]]] = None,
+    ) -> "PandasDataframe":
+        """
+        Replace the row and column labels with the specified new labels.
+
+        Parameters
+        ----------
+        new_row_labels : dict or callable, default: None
+            Mapping from old row labels to new labels.
+        new_col_labels : dict or callable, default: None
+            Mapping from old col labels to new labels.
+        level : int, default: None
+            Level whose row labels to replace.
+
+        Returns
+        -------
+        PandasDataframe
+            A new PandasDataframe with the new row and column labels.
+
+        Notes
+        -----
+        If level is not specified, the default behavior is to replace row labels in all levels.
+        """
+        new_index = self.index.copy()
+
+        def swap_labels(label_dict):
+            return (
+                lambda label: label_dict.get(label, label)
+                if isinstance(label_dict, dict)
+                else label_dict(label)
+            )
+
+        def swap_labels_levels(index_tuple):
+            return tuple(
+                new_row_labels.get(label, label)
+                if isinstance(new_row_labels, dict)
+                else new_row_labels(label)
+                for label in index_tuple
+            )
+
+        if new_row_labels:
+            swap_row_labels = swap_labels(new_row_labels)
+            if isinstance(self.index, pandas.MultiIndex):
+                if level is None:
+                    new_index = new_index.map(swap_labels_levels)
+                else:
+                    new_index.set_levels(
+                        new_index.levels[level].map(swap_row_labels), level
+                    )
+            else:
+                new_index = new_index.map(swap_row_labels)
+        new_cols = self.columns.copy()
+        if new_col_labels:
+            new_cols = new_cols.map(swap_labels(new_col_labels))
+
+        def map_fn(df):
+            return df.rename(index=new_row_labels, columns=new_col_labels, level=level)
+
+        new_parts = self._partition_mgr_cls.map_partitions(self._partitions, map_fn)
+        return self.__constructor__(
+            new_parts,
+            new_index,
+            new_cols,
+            self._row_lengths,
+            self._column_widths,
+            self._dtypes,
+        )
+
+    def sort_by(
```

Comment:
again, what's the point of overriding a parent method but without actually implementing it?

Reply:
@devin-petersohn asked us to include it for completeness? We can remove it if that's better.

Reply:
So that we don't run into an error trying to instantiate an abstract class. (More info: https://github.com/modin-project/modin/pull/3717#discussion_r769155299)